commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/UriParser.java

```diff
@@ -89,7 +90,7 @@ private boolean readNonSeparator() {
                 cursor++;
                 return true;
             }
-            final String sub = path.substring(cursor + 1, cursor + 3);
+            final String sub = path.substring(cursor, cursor + 3);

```

Comment:
Should we use constants instead of magic numbers here? This would help maintenance in the future IMO.

Reply:
I've copied the `final String sub = path.substring(cursor, cursor + 3);` from the existing `readDot();` and `readSeparator();`
Not that on of my next commit, I'll remove this substring as it is a performance bottleneck. My local branch shows a 3x performance improvement.